equinix/resource_network_device_acc_test.go

```diff
@@ -823,6 +826,64 @@ func TestAccNetworkDevice_CGENIX_HA_Self_BYOL(t *testing.T) {
+		"device-type_code":                   "PA-VM",
+		"device-package_code":                "VM100",
+		"device-notifications":               []string{"marry@equinix.com", "john@equinix.com"},
+		"device-term_length":                 1,
+		"device-version":                     "10.1.3",
+		"device-interface_count":             10,
+		"device-core_count":                  2,
+		"device-cluster_name":                fmt.Sprintf("%s-%s", tstResourcePrefix, randString(6)),
+		"device-node0_vendorConfig_enabled":  true,
+		"device-node0_vendorConfig_hostname": fmt.Sprintf("tf-%s", randString(6)),
+		"device-node0_license_token":         licenseToken.(string),
+		"device-node1_vendorConfig_enabled":  true,
+		"device-node1_vendorConfig_hostname": fmt.Sprintf("tf-%s", randString(6)),
+		"device-node1_license_token":         licenseToken.(string),
+		"sshkey-resourceName":                "test",
+		"sshkey-name":                        fmt.Sprintf("%s-%s", tstResourcePrefix, randString(6)),
+		"sshkey-public_key":                  "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCXdzXBHaVpKpdO0udnB+4JOgUq7APO2rPXfrevvlZrps98AtlwXXVWZ5duRH5NFNfU4G9HCSiAPsebgjY0fG85tcShpXfHfACLt0tBW8XhfLQP2T6S50FQ1brBdURMDCMsD7duOXqvc0dlbs2/KcswHvuUmqVzob3bz7n1bQ48wIHsPg4ARqYhy5LN3OkllJH/6GEfqi8lKZx01/P/gmJMORcJujuOyXRB+F2iXBVYdhjML3Qg4+tEekBcVZOxUbERRZ0pvQ52Y6wUhn2VsjljixyqeOdmD0m6DayDQgSWms6bKPpBqN7zhXXk4qe8bXT4tQQba65b2CQ2A91jw2KgM/YZNmjyUJ+Rf1cQosJf9twqbAZDZ6rAEmj9zzvQ5vD/CGuzxdVMkePLlUK4VGjPu7cVzhXrnq4318WqZ5/lNiCST8NQ0fssChN8ANUzr/p/wwv3faFMVNmjxXTZMsbMFT/fbb2MVVuqNFN65drntlg6/xEao8gZROuRYiakBx8= user@host",
+		"acl-resourceName":                   "acl-cluster",
+		"acl-name":                           fmt.Sprintf("%s-%s", tstResourcePrefix, randString(6)),
+		"acl-description":                    randString(50),
+	}
+
+	deviceResourceName := fmt.Sprintf("equinix_network_device.%s", context["device-resourceName"].(string))
+	clusterAclResourceName := fmt.Sprintf("equinix_network_acl_template.%s", context["acl-resourceName"].(string))
+	var primary ne.Device
+	var primaryAcl ne.ACLTemplate
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: newTestAccConfig(context).withDevice().withSSHKey().withACL().build(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccNeDeviceExists(deviceResourceName, &primary),
+					testAccNeDeviceAttributes(&primary, context),
+					testAccNeDeviceStatusAttributes(&primary, ne.DeviceStateProvisioned, ne.DeviceLicenseStateApplied),
+					testAccNeDeviceClusterAttributes(deviceResourceName),
+					testAccNeDeviceClusterNodeAttributes(primary.ClusterDetails, context),
+					testAccNetworkACLTemplateExists(clusterAclResourceName, &primaryAcl),
+					resource.TestCheckResourceAttrSet(deviceResourceName, "acl_template_id"),
+					testAccNeDeviceACL(clusterAclResourceName, &primary),
+				),
+			},
+		},
```

Comment:
Let's add an Import step.

It should look like this:

```suggestion
                        {
                                ResourceName:      deviceResourceName,
                                ImportState:       true,
                                ImportStateVerify: true,
                        },
		},
```

Reply:
Here's some background on Import - https://www.terraform.io/plugin/sdkv2/resources/import

Reply:
Fixed.